src/components/shared/transactionPriority/transactionPriority.js

```diff
@@ -126,16 +127,25 @@ const TransactionPriority = ({
           </Tooltip>
         </span>
         <div className={`${styles.prioritySelector} priority-selector`}>
-          {tokenRelevantPriorities.map((priority, index) => (
-            <button
-              key={`fee-priority-${index}`}
-              className={`${styles.priorityTitle} ${index === selectedPriority ? styles.priorityTitleSelected : ''} option-${priority.title}`}
-              onClick={onClickPriority}
-              value={index}
-            >
-              {priority.title}
-            </button>
-          ))}
+          {tokenRelevantPriorities.map((priority, index) => {
+            let disabled = false;
+            if (index === 3) {
+              disabled = !priority.value && !loadError;
+            } else if (index !== 0) {
+              disabled = !priority.value || loadError;
+            }
```

Comment:
I don't understand the above logic. Could you please briefly explain this? 😃

Reply:
@UsamaHameed I updated this so it it much readable. Is it more understandable now or do you still want and explaination?